### DISCUSSION TITLE
Lesson 6: Brownie FundMe

### DISCUSSION THREAD

**Discussion by ManiBharathiSS** [smartcontractkit/full-blockchain-solidity-course-py#1723] [Q&A] [answered]
I get this error when i run test with the network ganache-local. But i am not getting the error when i run the test with development network
![error_test](https://user-images.githubusercontent.com/110660107/193004135-059b0b89-6af7-4856-bfb0-cfd2142b3ba7.png)

**Comment by Niferu** [comment]
Hello @ManiBharathiSS 

The error pops because your `brownie-config.yaml` is probably setup wrong. Double check all addresses you have added under network you are testing it on, because  price feed you have pasted there for eth_usd might be the problem here as it should be following: `0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419` (you can get that on mainnet here: https://docs.chain.link/docs/data-feeds/price-feeds/addresses/?network=ethereum)

Hope this might help.

**Reply by ManiBharathiSS** [reply]
@Neftyr thanks for the reply,
ganache-local is a local blockchain network. So the mocks are deployed and it doesn't need eth_usd_price_feed. Only for the test nets and mainnet fork, eth_usd_price_feed needs to be added in brownie-config. So it doesn't solve the problem.
![networks](https://user-images.githubusercontent.com/110660107/193037513-a263000b-13ec-4636-b97a-eb5d810e6931.png)

Correct me if I am wrong.

This error pops up only when i run tests on ganache-local.

**Reply by Niferu** [reply]
Hello @ManiBharathiSS 
Sorry I have misunderstood your question then and I thought it is not working for you on mainnet fork network. In this case please share your `deploy()` and `deploy_mocks()` functions as your test code looks good. I'm definitely not pro and I'm still beginner but as far as I know most of the "Gas Estimation Fails" comes from incorrect price feeds.

**Comment by cromewar** [comment, marked as answer]
As @Neftyr 
Please share your deploy script, as `bad_actor` might not be getting correctly the account.

**Reply by ManiBharathiSS** [reply]
@cromewar @Neftyr  Here are the scripts for deploy( ) and deploy_mocks( )

![deploy](https://user-images.githubusercontent.com/110660107/193749653-5aafea3f-3aba-44cd-ad67-c83c4365ffe5.png)
![getacc](https://user-images.githubusercontent.com/110660107/193749649-19a5990b-633c-46a6-89ee-dd67955da297.png)
![mock](https://user-images.githubusercontent.com/110660107/193749642-f7a35383-f955-4143-8f51-11625aea473e.png)

**Reply by cromewar** [reply]
Please also share the whole test script on which you are getting the project.

**Reply by ManiBharathiSS** [reply]
@cromewar @Neftyr 
![test](https://user-images.githubusercontent.com/110660107/194320962-a873c208-2d6e-4ffa-86d3-f87da37217ca.png)

**Reply by cromewar** [reply]
ok you have bad actor as `bad_actor = accounts.add()` you can change it to `accounts[2]` and try again.
Does this works?
Or  you can change the `get_account()` function to be able to receive the index like this:

```Python
def get_account(index=None, id=None):
    # accounts[0]
    # accounts.add("env")
    # accounts.load("id")
    if index:
        return accounts[index]
    if id:
        return accounts.load(id)
    if (
        network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS
        or network.show_active() in FORKED_LOCAL_ENVIRONMENTS
    ):
        return accounts[0]
    return accounts.add(config["wallets"]["from_key"])
```

**Reply by ManiBharathiSS** [reply]
I get the same error
![Screenshot_2022-10-07_12-53-40](https://user-images.githubusercontent.com/110660107/194496327-158dc7eb-0201-4fcf-9dba-60a2bafe0e47.png)

**Reply by Niferu** [reply]
Hi @ManiBharathiSS

Your codes looks good and when I applied them to my code it works for me, so I would suggest you copying repository and running that code to see if issue is solved, so we can confirm there is everything ok with brownie. Please also send full consol message after running `brownie test -s -k test_only_owner_can_withdraw` command.

**Reply by cromewar** [reply]
Also on my side, I cloned the repo and the code runs fine... please can you follow @Neftyr suggestion @ManiBharathiSS?
